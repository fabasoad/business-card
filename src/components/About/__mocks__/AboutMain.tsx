import * as React from 'react'

export default function AboutMain() {
  return (
    <div data-testid="AboutMain"></div>
  )
}
