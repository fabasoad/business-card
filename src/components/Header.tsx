import * as React from 'react'
import { Trans, useTranslation } from 'react-i18next'

export default function Header() {
  const { t } = useTranslation()
  return (
    <header className="header" id="header">
      <div className="header__text">
        <h1><Trans t={t}>business-card-header-title</Trans></h1>
        <h5 className="header__text-subtitle">{t('business-card-header-occupation')}</h5>
        <a className="btn btn-primary btn-lg" href="#about">{t('business-card-header-about-me')}</a>
        <a className="btn btn-primary btn-lg" href="#portfolio">{t('business-card-header-portfolio')}</a>
      </div>
    </header>
  )
}
