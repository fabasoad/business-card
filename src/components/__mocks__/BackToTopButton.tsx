import * as React from 'react'

export default function BackToTopButton() {
  return (
    <div data-testid="BackToTopButton"></div>
  )
}
