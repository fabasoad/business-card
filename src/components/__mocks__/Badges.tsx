import * as React from 'react'

export default function Badges() {
  return (
    <div data-testid="Badges"></div>
  )
}
