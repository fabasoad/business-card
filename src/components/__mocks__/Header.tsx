import * as React from 'react'

export default function Header() {
  return (
    <div data-testid="Header"></div>
  )
}
