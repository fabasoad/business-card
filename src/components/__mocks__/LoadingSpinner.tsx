import * as React from 'react'
import { LoadingSpinnerProps } from '../LoadingSpinner'

export default function LoadingSpinner({ background, color }: LoadingSpinnerProps) {
  return (
    <div data-testid={`LoadingSpinner-${background}-${color}`}></div>
  )
}
