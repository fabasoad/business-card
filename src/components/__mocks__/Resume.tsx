import * as React from 'react'

export default function Resume() {
  return (
    <div data-testid="Resume"></div>
  )
}
