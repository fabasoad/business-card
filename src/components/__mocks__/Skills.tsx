import * as React from 'react'

export default function Skills() {
  return (
    <div data-testid="Skills"></div>
  )
}
