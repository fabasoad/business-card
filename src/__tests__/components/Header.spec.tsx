/// <reference types="jest" />
import * as React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import Header from '../../components/Header'
import { useTranslation } from '../__mocks__/react-i18next'

let tMock

beforeAll(() => {
  tMock = useTranslation().t
})

test('should render Header correctly', () => {
  const wrapper: ShallowWrapper = shallow(<Header />)
  expect(wrapper).toMatchSnapshot()
  expect(tMock).toHaveBeenCalledTimes(3)
  new Array<string>(
    'business-card-header-occupation',
    'business-card-header-about-me',
    'business-card-header-portfolio'
  ).forEach((n, i) => expect(tMock).toHaveBeenNthCalledWith(i + 1, n))
  expect(
    wrapper.find('Trans').childAt(0).text()
  ).toBe('business-card-header-title')
})
