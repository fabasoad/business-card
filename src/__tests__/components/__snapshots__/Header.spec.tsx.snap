// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header
  className="header"
  id="header"
>
  <div
    className="header__text"
  >
    <h1>
      <Trans
        t={
          [MockFunction] {
            "calls": Array [
              Array [
                "business-card-header-occupation",
              ],
              Array [
                "business-card-header-about-me",
              ],
              Array [
                "business-card-header-portfolio",
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": "business-card-header-occupation",
              },
              Object {
                "type": "return",
                "value": "business-card-header-about-me",
              },
              Object {
                "type": "return",
                "value": "business-card-header-portfolio",
              },
            ],
          }
        }
      >
        business-card-header-title
      </Trans>
    </h1>
    <h5
      className="header__text-subtitle"
    >
      business-card-header-occupation
    </h5>
    <a
      className="btn btn-primary btn-lg"
      href="#about"
    >
      business-card-header-about-me
    </a>
    <a
      className="btn btn-primary btn-lg"
      href="#portfolio"
    >
      business-card-header-portfolio
    </a>
  </div>
</header>
`;
